app/controllers/payments_controller.rb

```diff
@@ -71,33 +70,101 @@ def sent
     end
   end
 
-  def mark_received
-    @ticket_request = TicketRequest.find(permit_params[:ticket_request_id])
-    return redirect_to root_path unless @ticket_request.can_view?(current_user)
+  private
 
-    @payment = Payment.where(ticket_request_id: @ticket_request.id,
-                             status: Payment::STATUS_IN_PROGRESS)
-                      .first
+  def ticket_request_id
+    permitted_params[:ticket_request_id]
+  end
 
-    if @payment
-      @payment.mark_received
-      @payment.ticket_request.mark_complete
-      PaymentMailer.payment_received(@payment).deliver_now
-      redirect_to :back
+  def set_payment
+    Rails.logger.debug { "PaymentsController: set_payment params: #{params}" }
+    if @ticket_request.present?
+      @payment = @ticket_request.payment
+    elsif permitted_params[:payment_intent] || permitted_params[:stripe_payment_id]
+      strip_id = permitted_params[:payment_intent] || permitted_params[:stripe_payment_id]
+      @payment = Payment.where(stripe_payment_id: strip_id).first
+    elsif permitted_params[:ticket_request_id].is_a?(Numeric)
```

Comment:
`permitted_params[:ticket_request_id]` most likely returns a string. You'd need to do something like this:

```ruby
value = Integer(foo) rescue nil
if value.nil?
  puts "value is not an integer"
else
  puts "value is an integer"
end
```